### PR TITLE
Change case-insensitive Json default in SignalR

### LINKED ISSUE #10774: Make SignalR JSON protocol case-insensitive by default
In https://github.com/aspnet/AspNetCore/issues/10723 MVC changed it's default deserialization behavior to be case-insensitive, to match prior behavior in Newtonsoft.Json. We should do the same in SignalR.

Should be a simple matter of changing https://github.com/aspnet/AspNetCore/blob/670865d2a8997602ca2733cf15a9b74fbea399f9/src/SignalR/common/Protocols.Json/src/Protocol/JsonHubProtocol.cs#L779

And, of course, updating any tests ;).

### PR DESCRIPTION
**Pull request by BrennanConroy**
Fixes #10774

### CODE REVIEW

File: src/SignalR/common/Protocols.Json/src/Protocol/JsonHubProtocol.cs [hunk 1]
```diff
@@ -776,7 +776,7 @@ internal static JsonSerializerOptions CreateDefaultSerializerSettings()
             options.IgnoreNullValues = false;
             options.IgnoreReadOnlyProperties = false;
             options.PropertyNamingPolicy = JsonNamingPolicy.CamelCase;
-            options.PropertyNameCaseInsensitive = false;
+            options.PropertyNameCaseInsensitive = true;
```
Comment (rynowak) [inline src/SignalR/common/Protocols.Json/src/Protocol/JsonHubProtocol.cs]: I can confirm that this is the thing to change.

**Review by rynowak — APPROVED** [review]

File: src/SignalR/common/SignalR.Common/test/Internal/Protocol/JsonHubProtocolTests.cs [hunk 2]
```diff
@@ -101,6 +101,23 @@ public void MagicCast()
...
+            var input = Frame("{\"type\":2,\"invocationId\":\"123\",\"item\":{\"StrIngProp\":\"SignalR!\",\"DoublePrOp\":6.2831853071,\"IntProp\":43,\"DateTimeProp\":\"2017-04-11T00:00:00Z\",\"NuLLProp\":null,\"ByteARRProp\":[1,2,3]}}");
+
+            var binder = new TestBinder(null, typeof(TemporaryCustomObject));
+            var data = new ReadOnlySequence<byte>(Encoding.UTF8.GetBytes(input));
+            JsonHubProtocol.TryParseMessage(ref data, binder, out var message);
+
+            var streamItemMessage = Assert.IsType<StreamItemMessage>(message);
+            Assert.Equal(new TemporaryCustomObject()
+            {
+                ByteArrProp = new byte[] { 1, 2, 3 },
```
Comment (mikaelm12) [inline src/SignalR/common/SignalR.Common/test/Internal/Protocol/JsonHubProtocolTests.cs]: Why don't you check any of the other properties here?

**Review by mikaelm12 — APPROVED** [review]

File: src/SignalR/common/SignalR.Common/test/Internal/Protocol/JsonHubProtocolTests.cs (hunk 2 above)
Comment (BrennanConroy) [reply src/SignalR/common/SignalR.Common/test/Internal/Protocol/JsonHubProtocolTests.cs]: I guess there is no harm in changing all properties, I'll update :+1:

**Review by JamesNK — APPROVED** [review]

**Comment by BrennanConroy** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=214210&view=logs

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/2625

**Pull request merged**